pkg/test/istio.io/tasks/security/authn-policy/authn-policy_test.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+package tests
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/istio.io/examples"
+)
+
+var (
+	ist istio.Instance
+)
+
+func TestMain(m *testing.M) {
+	framework.NewSuite("authn-policy", m).
+		SetupOnEnv(environment.Kube, istio.Setup(&ist, setupConfig)).
+		RequireEnvironment(environment.Kube).
+		Run()
+}
+
+func setupConfig(cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	// This is redundant, but setting it explicitly to match the docs as it's explicitly required
+	// in the docs.
+	cfg.Values["global.mtls.enabled"] = "false"
+}
+
+// https://preliminary.istio.io/docs/tasks/security/authn-policy/
+// https://github.com/istio/istio.io/blob/master/content/docs/tasks/security/authn-policy/index.md
+func TestAuthnPolicy(t *testing.T) {
+	ex := examples.New(t, "Setup")
+
+	ex.AddScript("", "create-namespaces.sh", examples.TextOutput)
+	ex.AddFile("foo", "samples/httpbin/httpbin.yaml")
+	ex.AddFile("foo", "samples/sleep/sleep.yaml")
+	ex.AddFile("bar", "samples/httpbin/httpbin.yaml")
+	ex.AddFile("bar", "samples/sleep/sleep.yaml")
+	ex.AddFile("legacy", "samples/httpbin/httpbin.yaml")
+	ex.AddFile("legacy", "samples/sleep/sleep.yaml")
+
+	// This is missing from the docs, but it is necessary before continuing.
```

Comment:
The idea of having a validation function for scripts (i.e. should return error, should not return error, should have text...) was suggested yesterday. Might be a nice to have that idea for files as well. Waiting for containers to be ready and verifying reachability are likely common activities.